docs/rfc/rfc-026-p2p-bad-peers-checktx.md

```diff
@@ -0,0 +1,436 @@
+to disconnect from peers after they send us transactions that constantly fail `CheckTx`.  While 
+the actual implementation of an additional response code for `CheckTx` is straight forward there 
+are certain correctness aspects to consider. The questions to answer, along with identified risks will be outlined in 
+the discussion. 
+
+### Existing issues and concerns
+
+Before diving into the details, we collected a set of issues opened by various users, arguing for 
+this behaviour and explaining their needs. 
+
+- Celestia: [blacklisting peers that repeatedly send bad tx](https://github.com/celestiaorg/celestia-core/issues/867)
+  and investigating how [Tendermint treats large Txs](https://github.com/celestiaorg/celestia-core/issues/243)
+- BigChainDb: [Handling proposers who propose bad blocks](https://github.com/bigchaindb/BEPs/issues/84)
+- IBC relayers: Nodes allow transactions with a wrong `minGas` transaction and gossip them, and other nodes keep rejecting them. (Problem seen with relayers)
+
+**Acceptable duplicate transactions**
+
+Banning peers was also mentioned within [IBC-go](https://github.com/cosmos/ibc-go/issues/853#issuecomment-1032211020). However,the crux of the issue is preventing transactions with duplicate payload. While this is indeed undesired behaviour, this is not considered behaviour that should lead to banning a peer or even disconnecting from him. Duplicate transactions are in this case prevented using an application-specific solution. 
+
+### Current state of mempool/p2p interaction
+
+Transactions received from a peer are handled within the [`Receive`](https://github.com/tendermint/tendermint/blob/ff0f98892f24aac11e46aeff2b6d2c0ad816701a/mempool/v0/reactor.go#L158) routine. 
+
+Currently, the mempool triggers a disconnect from a peer in the case of the following errors:
+
+  - [Unknown message type](https://github.com/tendermint/tendermint/blob/ff0f98892f24aac11e46aeff2b6d2c0ad816701a/mempool/v0/reactor.go#L184)
+
+However, disconnecting from a peer is not the same as banning the peer. The p2p layer will close the connecton but 
+the peer can reconnect without any penalty, and if it as a persistent peer, a reconnect will be initiated
+from the node. 
+
+### Current support for peer banning
+
+The p2p layer implements banning peers by marking them
+as bad and removing them from the list of peers to connect to for *at least* a predefined amount of time. This is done by calling the `MarkBad` routine implemented by the `Switch`. If the node does not set the amount of time to be banned, a default value is used. 
+Note that the timing parameter sets the lower bound for when a peer will be unbanned. 
+But the p2p layer will only try to connect to banned peers if the node is not sufficiently connected. Thus the node has no
+explicit control on when a reconnect attempt will be triggered.
+
+The application can blacklist peers via ABCI if the 
+[`filterPeers`](../../spec/abci/abci%2B%2B_app_requirements.md#peer-filtering) 
```

Comment:
Are you sure that a flag is needed?

Reply:
Are you sure we should refer to main rather than to the relative path? In  case the spec changes on main at some point..although I guess the RFCs will always be on main?